### PR TITLE
Produce a real error message when an annotation value is not a string

### PR DESCRIPTION
**Pull request by msullivan**

### CODE REVIEW

**Review by elprans — APPROVED** [review]

File: edb/schema/annos.py [hunk 1]
```diff
@@ -290,8 +292,10 @@ def _cmd_tree_from_ast(
             astnode.value, schema=schema)
 
         if not isinstance(value, str):
-            raise ValueError(
-                f'unexpected value type in annotation: {value!r}')
+            raise errors.InvalidValueError(
+                'annotation values must be strings',
```
Comment (1st1) [inline edb/schema/annos.py:296]:
```suggestion
                'annotation values must be strings, got ${typename}',
```

File: edb/schema/annos.py [hunk 2]
```diff
@@ -366,8 +370,10 @@ def _cmd_tree_from_ast(
                 astnode.value, schema=schema)
 
             if not isinstance(value, str):
-                raise ValueError(
-                    f'unexpected value type in AnnotationValue: {value!r}')
+                raise errors.InvalidValueError(
+                    'annotation values must be strings',
```
Comment (1st1) [inline edb/schema/annos.py:374]: ditto

File: edb/schema/annos.py [hunk 3]
```diff
@@ -282,18 +282,23 @@ def _cmd_tree_from_ast(
...
         assert isinstance(cmd, CreateAnnotationValue)
         annoname = sn.shortname_from_fullname(cmd.classname)
 
-        value = qlcompiler.evaluate_ast_to_python_val(
+        value, ir = qlcompiler.evaluate_ast_to_python_val_and_ir(
             astnode.value, schema=schema)
 
-        if not isinstance(value, str):
+        if (
+            not isinstance(ir.stype, s_scalar.ScalarType)
```
Comment (elprans) [inline edb/schema/annos.py:296]: Use `ir.stype.is_scalar()` and then you don't need an import?

Comment (msullivan) [reply edb/schema/annos.py:296]: Actually we don't even need to check that it's a scalar, do we?

Comment (elprans) [reply edb/schema/annos.py]: We don't.

**Pull request merged**